python/src/cloudstorage/cloudstorage_api.py

```diff
@@ -273,6 +275,185 @@ def listbucket(path_prefix, marker=None, prefix=None, max_keys=None,
+  (this will only need to happen once, since the file_list size restriction is 1024 = 32 * 32)
+  '''
+  temp_list = []  # temporary storage for the filenames that store the merged segments of 32
+  if len(file_list) > 32:
+    temp_file_counter = 0
+    segments_list = [file_list[i:i + 32] for i in range(0, len(file_list), 32)]
+    file_list = []
+    for segment in segments_list:
+      if len(segment) > 1:
+        temp_file_name = destination_file + temp_file_suffix + str(temp_file_counter)
+        compose(segment, temp_file_name, content_type, retry_params, _account_id)
+        file_list.append({'file_name': temp_file_name.replace(bucket, '', 1)})
+        temp_file_counter += 1
+        temp_list.append(temp_file_name)
+      else:
+        file_list.append(segment[0])
+  # There will always be 32 or less files to merge at this point
+  compose(file_list, destination_file, content_type, retry_params, _account_id)
+  # grab all temp files that were created during the merging of segments of 32
+  temp_list = listbucket(destination_file + temp_file_suffix)
+  # delete all the now-unneeded temporary merge-files for the segments of 32 (if any)
+  for item in temp_list:
+    try:
+      delete(item.filename)
+    except errors.NotFoundError:
+      pass
+
+def file_exists(destination):
+  '''Checks if a file exists.
+    Tries to do a stat check on the file.
+    If it succeeds returns True otherwise false
+
+    Args:
+      destination: Full path to the file (ie. /bucket/object) w/ leading slash
+    Returns:
+      True if the file is accessible otherwise False
+    '''
+  try:
+    stat(destination)
+    return True
+  except errors.NotFoundError:
+    return False
+
+def _validate_compose_list(destination_file, file_list, content_type, number_of_files=32):
+  '''
+    Validates the compose list and extracts needed information from it
+    Args:
+      destination: Full path to the file (ie. /bucket/object),
+      file_list: list of files to compose, see compose for details,
+      content_type: content type specified by the user,
+      number_of_files=32: maximum number of files allowed in the list
+    Returns:
+      list_of_files: Ready to use dict version of the list,
+      bucket: bucket name extracted from the file paths,
+      content_type: either the specified content_type or one devined from the first file
+  '''
+  common.validate_file_path(destination_file)
+  bucket = '/' + destination_file.split('/')[1] + '/'
+  if not isinstance(file_list, list):
+    raise TypeError('file_list must be a list of strings or dictionaries')
+  list_len = len(file_list)
+  if list_len > number_of_files:
+    raise ValueError(
+          'Compose attempted to create composite with too many'
+           '(%i) components; limit is (%i).' % (list_len, number_of_files))
+  if list_len <= 1:
+    raise ValueError('Compose operation requires at least two components; %i provided.' % list_len)
+
+  if isinstance(file_list[0], str):
```

Comment:
If you prefer one list (instead of 2 for files and metadata) then i think we should accept mixed values in the list.